### PR TITLE
Binary scanner

### PR DESCRIPTION
**Pull request by andymckay**
* requires #635 
* look for bad file extensions
* look for bad magic numbers
* since the null scanner wasn't being used, but instead was the fallback to search for binary files, changed its name to binary scanner
* since the hidden scanner was basically just scanning by filename, i thought filename scanner would be more appropriate

### CODE REVIEW

**Comment by andymckay** [comment]
Will need rebasing with #635 when that lands.

File: src/const.js [hunk 1]
```diff
@@ -116,3 +116,21 @@ export const MAX_FILE_SIZE_MB = 100;
 
 export const HIDDEN_FILE_REGEX = /^__MACOSX\//;
 export const FLAGGED_FILE_REGEX = /thumbs.db$|.DS_Store$|.orig$|.old$|\~$/i;
+
+export const FLAGGED_FILE_EXTENSION_REGEX =
+  /.exe$|.dll$|.dylib$|.so$|.sh$|.class$|.swf$/i;
```
Comment (tofumatt) [inline src/const.js]: Do the dots need escaping (eg `\.exe`)?

File: src/scanners/binary.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    this.linterMessages.push(
+      Object.assign({}, messages.FLAGGED_FILE_TYPE, {
+        type: constants.VALIDATION_NOTICE,
+        file: this.filename,
+      })
+    );
+  }
+
+  scan() {
+    return new Promise((resolve) => {
```
Comment (tofumatt) [inline src/scanners/binary.js]: You can instead return `return Promise.resolve(this.linterMessages);` at the bottom and avoid returning the function. We've been trying to do that in this repo.

File: tests/scanners/test.binary.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import BinaryScanner from 'scanners/binary';
+import {FLAGGED_FILE_MAGIC_NUMBERS} from 'const';
```
Comment (tofumatt) [inline tests/scanners/test.binary.js]: This is a major nitpick but we usually put spaces around grouped imports like this:

```js
import { FLAGGED_FILE_MAGIC_NUMBERS } from 'const';
```

**Comment by tofumatt** [comment]
r+wc (and obviously #635 being merged 😄 ).

File: src/const.js (hunk 1 above)
Comment (andymckay) [inline src/const.js]: Actually they should, interestingly in FLAGGED_FILE_REGEX too.

File: src/scanners/binary.js (hunk 2 above)
Comment (andymckay) [inline src/scanners/binary.js]: Ah cool, thanks

File: tests/scanners/test.binary.js (hunk 3 above)
Comment (andymckay) [inline tests/scanners/test.binary.js]: Rightio :)

**Comment by andymckay** [comment]
fixes #56

**Pull request merged**